lib/interfaces/mongoose-options.interface.ts

```diff
@@ -18,13 +18,15 @@ export interface MongooseOptionsFactory {
     | MongooseModuleOptions;
 }
 
+export type MongooseModuleOptionsFactory = Omit<MongooseModuleOptions, 'connectionName'>
```

Comment:
The name here is somewhat confusing as this isn't a factory but rather options object that must be passed **into** a factory function.

Reply:
any idea for better naming? `MongooseModuleFactoryOptions`?

Reply:
sounds good

Reply:
done